sympy/core/basic.py

```diff
@@ -905,6 +905,21 @@ def subs(self, *args, **kwargs):
             return rv.xreplace(reps)
         else:
             rv = self
+            if len(sequence) > 1:
+                try:
+                    from sympy.utilities.iterables import topological_sort
+                    substitution = args[0]
+                    topo_sorted_subs = topological_sort(((list(set(substitution.keys() +
+                                                                   substitution.values())),
+                                                          sequence)))
+                    sequence = [(item, substitution[item])
+                                for item in topo_sorted_subs
+                                if item in substitution.keys()]
+                except:
```

Comment:
Actually, I can throw something like that TopologySortException instead ValueError  to catch error from topological_sort.